packages/infra/lib/ihe-stack/ihe-db-construct.ts

```diff
@@ -0,0 +1,105 @@
+import { Aspects, CfnOutput } from "aws-cdk-lib";
+import { BackupResource } from "aws-cdk-lib/aws-backup";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as rds from "aws-cdk-lib/aws-rds";
+import * as secret from "aws-cdk-lib/aws-secretsmanager";
+import { Construct } from "constructs";
+import { IHEGatewayProps } from "../../config/ihe-gateway-config";
+import { EnvType } from "../env-type";
+import { DailyBackup } from "../shared/backup";
+import { addDBClusterPerformanceAlarms } from "../shared/rds";
+import { isProdEnv } from "../shared/util";
+
+// export interface IHEGatewayAlarmThresholds {
+//   masterCpuUtilization?: number;
+//   cpuUtilization?: number;
+// }
+export interface IHEDatabaseConstructProps {
+  env: EnvType;
+  config: IHEGatewayProps;
+  vpc: ec2.IVpc;
+  alarmAction?: SnsAction | undefined;
+}
+
+const id = "IHEDatabase";
+
+export default class IHEDBConstruct extends Construct {
+  public readonly secret: secret.ISecret;
+  public readonly server: rds.IDatabaseCluster;
+
+  constructor(scope: Construct, props: IHEDatabaseConstructProps) {
+    super(scope, `${id}Construct`);
+
+    const { vpc, config, alarmAction } = props;
+
+    const dbClusterName = "ihe-db";
+    const dbName = config.rds.dbName;
+    const dbUsername = config.rds.userName;
+    const dbPasswordSecret = new secret.Secret(this, `${id}Password`, {
+      secretName: `${id}Password`,
+      generateSecretString: {
+        excludePunctuation: true,
+        includeSpace: false,
+      },
+    });
+    const dbCreds = rds.Credentials.fromPassword(dbUsername, dbPasswordSecret.secretValue);
+    this.secret = dbPasswordSecret;
+
+    const dbEngine = rds.DatabaseClusterEngine.auroraPostgres({
+      version: rds.AuroraPostgresEngineVersion.VER_14_7,
+    });
+    // TODO 1377 validate that it copies the props from the default group
+    const parameterGroup = new rds.ParameterGroup(this, "FHIR_DB_Params", {
+      engine: dbEngine,
+      parameters: {
+        log_min_duration_statement: config.rds.minSlowLogDurationInMs.toString(),
+      },
+    });
+    const dbCluster = new rds.DatabaseCluster(this, `${id}DB`, {
+      engine: dbEngine,
+      instanceProps: {
+        vpc,
+        instanceType: new ec2.InstanceType("serverless"),
+      },
+      credentials: dbCreds,
+      defaultDatabaseName: dbName,
+      clusterIdentifier: dbClusterName,
+      storageEncrypted: true,
+      parameterGroup,
+      cloudwatchLogsExports: ["postgresql"],
+    });
+    this.server = dbCluster;
```

Comment:
just a general food for thought - are we sure we want a separate DB, and not just leverage the OSS one we already have?

Reply:
Leaving as is for now until we come up with a definition [on Slack](https://metriport.slack.com/archives/C04DMKE9DME/p1707579501239229?thread_ts=1707534426.135549&cid=C04DMKE9DME).

I can move it to internal on the next PR if we decide to do it. 👍